### PR TITLE
store: use delete_file_in_range to destroy region.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing @zhangjinpeng1987 @overvenus

### CODE REVIEW

File: src/raftstore/store/engine.rs [hunk 1]
```diff
@@ -240,11 +240,29 @@ pub trait Mutable: Writable {
 impl Mutable for DB {}
 impl Mutable for WriteBatch {}
 
+// `delete_in_range` fast deletes date in range [start_key, end_key).
+// It uses rocksdb `delete_file_in_range` first, then scans the left keys and
+// uses WriteBatch to deletes them.
+// Note: this function is dangerous and not guarantees consistence. If you call
+// delete_file_in_range successfully but commit following WriteBatch failed,
+// some keys are really deleted and can't be recovered.
+pub fn delete_in_range(db: &DB, wb: &WriteBatch, start_key: &[u8], end_key: &[u8]) -> Result<()> {
```
Comment (BusyJay) [inline src/raftstore/store/engine.rs]: Is `WriteBatch` here necessary?

Comment (siddontang) [inline src/raftstore/store/engine.rs]: yes, delete_file_in_range can't guarantee deleting all data in [start, end), we must scan to delete left keys.

File: src/raftstore/store/peer_storage.rs [hunk 2]
```diff
@@ -429,17 +429,28 @@ impl PeerStorage {
...
-                                      Ok(true)
-                                  }));
+
+        for (start_key, end_key) in self.region_key_ranges() {
+            // meta range only has little keys, no need use unsafe delete.
+            if !start_key.starts_with(keys::REGION_META_PREFIX_KEY) {
+                // TODO:
+                // 1. need to handle `delete_in_range` successfully but following
+                // WriteBatch failed.
+                // 2. need to care other CF?
```
Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: I think all cf should call `delete_in_range`.

File: src/raftstore/store/engine.rs (hunk 1 above)
Comment (BusyJay) [inline src/raftstore/store/engine.rs]: Why not just call `db.delete`?

File: src/raftstore/store/peer_storage.rs (hunk 2 above)
Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: I don't know, if data in other cf is small,  no need. 

@disksing

File: src/raftstore/store/engine.rs (hunk 1 above)
Comment (siddontang) [inline src/raftstore/store/engine.rs]: we may still have many keys left, delete every key is not efficient. 
e.g, if the range crosses two SST, the `delete_file_in_range` does nothing, so we will degrade origin deleting way.

File: src/raftstore/store/peer_storage.rs (hunk 2 above)
Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: If `delete_in_range` can finish fast, then it doesn't matter whether the data is small or not. It doesn't delete extra data and keep the code simple.

File: src/raftstore/store/engine.rs (hunk 1 above)
Comment (BusyJay) [inline src/raftstore/store/engine.rs]: I think in this case we can buffer the deletion, instead of putting the whole region into the memory.

**Pull request closed without merging**